srml/staking/src/lib.rs

```diff
@@ -1533,12 +1575,32 @@ impl <T: Trait> OnOffenceHandler<T::AccountId, session::historical::Identificati
 				remaining_imbalance.subsume(slashed_amount);
 			}
 		}
+		<EraSlashJournal<T>>::insert(era_now, journal);
 
 		// Handle the rest of imbalances
 		T::Slash::on_unbalanced(remaining_imbalance);
 	}
 }
 
+/// Filter historical offences out and only allow those from the current era.
+pub struct FilterHistoricalOffences<T, R> {
```

Comment:
Why not hook it up in node-runtime right away?

Reply:
I wanted to, but I didn't see that `ReportOffence` was used anywhere